src/notification-hub/azext_notification_hub/commands.py

```diff
@@ -7,59 +7,32 @@
-        g.custom_command('create', 'create_or_update_authorization_rule_notificationhubs_hub')
-        g.custom_command('delete', 'delete_authorization_rule_notificationhubs_hub')
-        g.custom_command('list-keys', 'list_keys_notificationhubs_hub')
-        g.custom_command('regenerate-keys', 'regenerate_keys_notificationhubs_hub')
-
-    with self.command_group('notification-hub credential', notificationhubs_notification_hubs, client_factory=cf_notification_hubs) as g:
-        g.custom_command('list', 'get_pns_credentials_notificationhubs_hub')
-        g.custom_command('gcm update', 'update_gcm_credential')
-        g.custom_command('apns update', 'update_apns_credential')
-        g.custom_command('wns update', 'update_wns_credential')
-        g.custom_command('mpns update', 'update_mpns_credential')
-        g.custom_command('adm update', 'update_adm_credential')
-        g.custom_command('baidu update', 'update_baidu_credential')
+    with self.command_group('notification-hub'):
+        from azext_notification_hub.custom import NotificationHubCreate, NotificationHubUpdate
+        self.command_table['notification-hub create'] = NotificationHubCreate(loader=self)
+        self.command_table['notification-hub update'] = NotificationHubUpdate(loader=self)
+
+    with self.command_group('notification-hub authorization-rule'):
+        from azext_notification_hub.custom import RuleRegenerateKeys
+        self.command_table['notification-hub authorization-rule regenerate-keys'] = RuleRegenerateKeys(loader=self)
+
+    with self.command_group('notification-hub namespace'):
+        from azext_notification_hub.custom import NamespaceCreate
+        self.command_table['notification-hub namespace create'] = NamespaceCreate(loader=self)
+
+    with self.command_group('notification-hub namespace authorization-rule'):
+        from azext_notification_hub.custom import NamespaceRuleCreate, NamespaceRuleRegenerateKeys
+        self.command_table['notification-hub namespace authorization-rule create'] = NamespaceRuleCreate(loader=self)
+        self.command_table['notification-hub namespace authorization-rule regenerate-keys'] = NamespaceRuleRegenerateKeys(loader=self)
+
+    with self.command_group('notification-hub credential'):
+        from azext_notification_hub.custom import BaiduUpdate, ApnsUpdate, MpnsUpdate, AdmUpdate, WnsUpdate, GcmUpdate
+        self.command_table['notification-hub credential gcm update'] = GcmUpdate(loader=self)
+        self.command_table['notification-hub credential adm update'] = AdmUpdate(loader=self)
+        self.command_table['notification-hub credential apns update'] = ApnsUpdate(loader=self)
+        self.command_table['notification-hub credential wns update'] = WnsUpdate(loader=self)
+        self.command_table['notification-hub credential mpns update'] = MpnsUpdate(loader=self)
+        self.command_table['notification-hub credential baidu update'] = BaiduUpdate(loader=self)
```

Comment:
Why not use the aaz generated update command directly?

Reply:
![image](https://github.com/Azure/azure-cli-extensions/assets/69238381/14a0944c-ccc4-41e3-baf3-512be9a87f35)